include/effectfactories.h

```diff
@@ -115,9 +115,7 @@ class EffectFactories
     void AddEffect(int effectNumber, const DefaultEffectFactory& defaultFactory, const JSONEffectFactory& jsonFactory)
     {
         defaultFactories.emplace_back(effectNumber, defaultFactory);
-
-        if (jsonFactories.count(effectNumber) == 0)
-            jsonFactories[effectNumber] = jsonFactory;
+        jsonFactories.try_emplace(effectNumber, jsonFactory);    

```

Comment:
I assume the possible outcomes are 0 or 1 for existing factories, but please vet my assumption.  I mean it's a map, so... try_emplace should do the trick!

Reply:
I believe try_emplace is atomic; there's a race condition in the first version because it's done in two steps.  Is that a real concern?  Not here, no.